### PR TITLE
PB-7099. SDK-Extend get groups method to allow searching by name

### PR DESCRIPTION
**Pull request by kmmanek**

### CODE REVIEW

File: sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        createdGroups.add(newGroup("sixth_Group"));
+    }
+
+    private void cleanUpTestData() {
+        for(Group group : createdGroups) {
+            eslClient.getGroupService().deleteGroup(group.getId());
+        }
+    }
+
+    public List<Group> getThirdGroup() {
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java]: Delete this method and use that variable immediately.

File: sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        return thirdGroup;
+    }
+
+    public List<Group> getLastThreeGroup() {
+        return lastThreeGroup;
+    }
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java]: Delete this method and use that variable immediately.

File: sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public List<Group> getAllGroup() {
+        return allGroup;
+    }
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java]: Delete this method and use that variable immediately.

File: sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+    public List<Group> getMissingGroup() {
+        return missingGroup;
+    }
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/examples/GroupRetrievalExample.java]: Delete this method and use that variable immediately.

File: sdk/src/main/java/com/silanis/esl/sdk/internal/UrlTemplate.java [hunk 5]
```diff
@@ -93,6 +93,7 @@
     public static final String GROUPS_ID_PATH = "/groups/{groupId}";
     public static final String GROUPS_MEMBER_PATH = "/groups/{groupId}/members";
     public static final String GROUPS_INVITE_PATH = "/groups/{groupId}/invite";
+    public static final String GROUPS_NAME_PATH = "/groups?name={groupName}";
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/internal/UrlTemplate.java]: Remove this line and use the addParam method.

File: sdk/src/main/java/com/silanis/esl/sdk/service/GroupService.java [hunk 6]
```diff
@@ -31,6 +28,21 @@ public GroupService( RestClient client, String baseUrl ) {
...
+    }
+
+    /**
+     * Retrieve group(s) belonging to the account by specifying complete (or part of) GroupName.
+     *
+     * @param groupName
+     * @return
+     */
+    public List<Group> getMyGroups( String groupName ) {
+        String path = template.urlFor( UrlTemplate.GROUPS_NAME_PATH ).replace( "{groupName}", groupName ).build();
```
Comment (SeungChoi) [inline sdk/src/main/java/com/silanis/esl/sdk/service/GroupService.java]: Use the addParam method.

File: tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+import static junit.framework.TestCase.assertTrue;
+
+public class GroupRetrievalExampleTest {
+
+    @Test
+    public void verifyResult() {
+        GroupRetrievalExample example = new GroupRetrievalExample();
+        example.run();
+
+        assertTrue(example.getThirdGroup().get(0).getName().equalsIgnoreCase("third_Group"));
```
Comment (SeungChoi) [inline tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java]: Do not use assertTrue in this case. Use assertThat and Matcher

File: tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        assertTrue(example.getThirdGroup().size() == 1);
```
Comment (SeungChoi) [inline tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java]: Do not use assertTrue in this case. Use assertThat and Matcher( hasSize in this case)

File: tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        assertTrue(example.getLastThreeGroup().size() == 3);
```
Comment (SeungChoi) [inline tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java]: Do not use assertTrue in this case. Use assertThat and Matcher( hasSize in this case)

File: tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+        assertTrue(example.getAllGroup().size() == 6);
```
Comment (SeungChoi) [inline tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java]: Do not use assertTrue in this case. Use assertThat and Matcher( hasSize in this case)

File: tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+        assertTrue(example.getMissingGroup().isEmpty());
```
Comment (SeungChoi) [inline tester/src/test/java/com/silanis/esl/sdk/examples/GroupRetrievalExampleTest.java]: Do not use assertTrue in this case. Use assertThat and Matcher

**Comment by kmmanek** [comment]
pushed update to the code.

**Pull request merged**